### DISCUSSION TITLE
Basic usage

### DISCUSSION THREAD

**Discussion by developer992** [O365/python-o365#1075] [General]
Hello,

i am a bit confused with this library as I am not too familiar with Microsoft's APIs.

Here's what I need to do, I need to upload a file to Sharepoint, and then call File preview on the Graph API https://learn.microsoft.com/en-us/graph/api/driveitem-preview?view=graph-rest-1.0 ... Can this be done using this library?

Many thanks!

Edit:

I've managed to authenticate but beyond that, i am not too sure how to proceed:

```
credentials = (CLIENT_ID, CLIENT_SECRET)
scopes = ['sharepoint_dl']
account = Account(credentials, scopes=scopes)
account.authenticate()
```

**Comment by alejcas** [comment]
The preview endpoint is not currently implemented.

**Comment by developer992** [comment]
Ok thanks, how about How do i upload a file?

**Comment by developer992** [comment]
"https://tenant.sharepoint.com/teams/testSite/_api/web/GetFolderByServerRelativeUrl('%20Shared%20Documents')/Files/add(url='a.txt',overwrite=true)"

Can I call this endpoint?

**Reply by alejcas** [reply]
please follow the readme, deep dive into the code or check other issues or examples

**Comment by developer992** [comment]
I have, that's all there is to it, this sentence :"
The SharePoint api is done but there are no docs yet. Look at the sharepoint.py file to get insights."

i'll figure it out, thanks anyway

**Comment by developer992** [comment]
Which class would i add this to, if i wanted to hack it 

Sharepoint, Drive, List, Site?

**Reply by alejcas** [reply]
You can get to Rome using many different roads, but get the Site and from there the document library
